scipy/spatial/_spherical_voronoi.py

```diff
@@ -0,0 +1,360 @@
+"""
+Spherical Voronoi Code
+
+.. versionadded:: 0.17.0
+
+"""
+#
+# Copyright (C)  Tyler Reddy, Ross Hemsley, Edd Edmondson,
+#                Nikolai Nowaczyk, Joe Pitt-Francis, 2015.
+#
+# Distributed under the same BSD license as Scipy.
+#
+
+import numpy as np
+import numpy.matlib
+import scipy
+import itertools
+from scipy._lib._version import NumpyVersion
+
+# Whether Numpy has stacked matrix linear algebra
+HAS_NUMPY_VEC_DET = (NumpyVersion(np.__version__) >= '1.8.0')
+
+__all__ = ['SphericalVoronoi']
+
+
+def determinant_fallback(m):
+    """
+    Calculates the determinant of m using Laplace expansion in the first row.
+    This function is used only as a fallback to ensure backwards compatibility
+    with Python 2.6.
```

Comment:
We were using `np.linalg.det` only in a previous built (see for instance b6724a9), which works perfectly fine in most unittest scenarios of `scipy`. However, in the unittest suite `Python 2.6` this throws the error: 

`LinAlgError: 3-dimensional array given. Array must be two-dimensional`

To my knowledge, the `np.linalg.det` function used in the version of numpy, which is in place during the `Python 2.6` unittests supports only `2x2`-matrices. Therefore, we introduced the check in line 21, to find out whether or not `np.linalg.det` from numpy can be used or if we have to use the fallback code. Determinants of `4x4` matrices are needed in the function `calc_circumcenters`. They are crucial to that function and this function is crucial to the entire module.

Therefore, I unfortunately don't see any way to remove the fallback code without breaking the `Python 2.6` unittest, although this is admittedly a bit ugly. :/  If you have any idea how to solve this problem more elegantly, please let us know! 

I totally agree that the docstring is misleading though. The problem is not the version of Python, but the version of numpy used in conjuction with Python 2.6 in the `Python 2.6` unittesting framework of scipy. Do you think it is ok, if I just change the docstring to ''to ensure backwards compatibility with numpy < 1.8.0" ?